### PR TITLE
Fix right alignment TWB longestLine

### LINKED ISSUE #49412: TextWidthBasis.longestLine + TextAlign.right is broken
```dart
import 'package:flutter/material.dart';

void main() => runApp(Center(
  child: Directionality(
    textDirection: TextDirection.ltr,
    child: DecoratedBox(
      decoration: BoxDecoration(
        border: Border.all(color: Colors.white),
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          Text(
            'LEFT ALIGNED, PARENT',
            textAlign: TextAlign.left,
            textWidthBasis: TextWidthBasis.parent,
          ),
          Text(
            'RIGHT ALIGNED, PARENT',
            textAlign: TextAlign.right,
            textWidthBasis: TextWidthBasis.parent,
          ),
          Text(
            'LEFT ALIGNED, LONGEST LINE',
            textAlign: TextAlign.left,
            textWidthBasis: TextWidthBasis.longestLine,
          ),
          Text(
            'RIGHT ALIGNED, LONGEST LINE',
            textAlign: TextAlign.right,
            textWidthBasis: TextWidthBasis.longestLine,
          ),
        ],
      ),
    ),
  ),
));
```
->
Screenshot_20200123-221628

cc @GaryQian

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Currently, there is a bug with `TextWidthBasis.longestLine` and `TextAlign.right` when aligned by a parent:

screen shot of bug

This PR fixes a bug in my implementation of TextWidthBasis (originally from https://github.com/flutter/flutter/pull/30988) where the wrong width is used for comparison.  This makes the broken case align based on its parent like the other cases:

Screen Shot 2020-04-07 at 2 47 30 PM

## Related Issues

Closes https://github.com/flutter/flutter/issues/49412

## Tests

I added the following tests:

  1. Render all combinations of textWidthBasis and textAlign, and ensure that their behavior is all the same and all obeys the alignment of their parent.

### CODE REVIEW

**Comment by goderbauer** [comment]
fly-by comment: The analyzer is unhappy.

**Comment by justinmc** [comment]
@goderbauer Thanks!

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -564,7 +564,7 @@ class TextPainter {
     _paragraph.layout(ui.ParagraphConstraints(width: maxWidth));
     if (minWidth != maxWidth) {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/painting/text_painter.dart:565]: Just want to make sure I'm understanding this part correctly: it shrink-wraps the `Paragraph` to `[minWidth, maxWidth]` if it occupies only one line?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM (if I understood line 565-569 correctly).

File: packages/flutter/lib/src/painting/text_painter.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/painting/text_painter.dart:565]: I think it does so even for multiline paragraphs if I'm reading the code correctly.

**Pull request merged**